### PR TITLE
ci: enable Nix caching

### PR DESCRIPTION
**Pull request by TobiasSN**
This will make CI runs go faster when some things don't need to be rebuilt. I replaced `cachix/install-nix-action` with `nixbuild/nix-quick-install-action`, since it's required by `nix-community/cache-nix-action`.

~~Do not merge yet, it still needs to be tested.~~ Ready pending approval.

### CODE REVIEW

**Comment by TobiasSN** [comment]
Blocked on #23, since the new action for installing Nix does not set up channels, and doing so should not be needed anyway.

**Comment by TobiasSN** [comment]
Rebased on `main` after merging #23

**Comment by TobiasSN** [comment]
Tested and successfully working. https://github.com/f-klubben/fnix/actions shows a time saving of 30-40 seconds on building Stregsystemet without changes. This will likely be more significant if and when we add more CI checks, build more packages on CI, and eventually automatically deploy to the server.

**Review by Mast3rwaf1z — APPROVED** [review]

**Pull request merged**
